VDF.Core/FFTools/FFProbeJsonReader.cs

```diff
@@ -18,114 +18,212 @@
+				Streams = new MediaInfo.StreamInfo[jsonInfo?.streams?.Count ?? 0]
+			};
 
-			var streams = new List<Dictionary<string, object>>();
-			var format = new Dictionary<string, object>();
+			if (jsonInfo != null && jsonInfo.streams != null) {
+				if (jsonInfo.format?.duration != null && TimeSpan.TryParse(jsonInfo.format.duration, out var duration))
+					/*
+					* Trim miliseconds here as we would have done it later anyway.
+					* Reasons are:
+					* - More user friendly
+					* - Allows an improved check against equality
+					* Cons are:
+					* - Not 100% accurate if you consider a difference of e.g. 2 miliseconds makes a duplicate no longer a duplicate
+					* - Breaking change at the moment of implementation as it doesn't apply to already scanned files
+					*/
+					info.Duration = duration.TrimMiliseconds();
 
-			var currentStream = -1;
+				long.TryParse(jsonInfo.format?.bit_rate, out var formatBitrate);
+		
+				for (int i = 0; i < info.Streams.Count(); i++) {
+					var stream = jsonInfo.streams[i];
+					info.Streams[i] = new MediaInfo.StreamInfo();
+					if (stream.bit_rate != null && long.TryParse(stream.bit_rate , out var bitrate))
+						info.Streams[i].BitRate = bitrate;
+					else if (stream.codec_type == "video")
+						info.Streams[i].BitRate = formatBitrate;
+						//Workaround if video stream bitrate is not set but in format
 
-			var currentObject = JsonObjects.None;
-			string? lastKey = null;
+					info.Streams[i].Width 			= stream.width;
+					info.Streams[i].Height 			= stream.height;
+					info.Streams[i].CodecName 		= stream.codec_name ?? "";
+					info.Streams[i].CodecLongName 	= stream.codec_long_name ?? "";
+					info.Streams[i].CodecType 		= stream.codec_type ?? "";
+					info.Streams[i].ChannelLayout 	= stream.channel_layout ?? "";
+					info.Streams[i].PixelFormat 	= stream.pix_fmt ?? "";
+					info.Streams[i].Default			= stream.disposition?.default_ ?? 0;
+					info.Streams[i].Index 			= stream.index.ToString();
+					
+					if (int.TryParse(stream.sample_rate, out var SampleRate))
+						info.Streams[i].SampleRate = SampleRate;
 
-			while (json.Read()) {
-				JsonTokenType tokenType = json.TokenType;
-				ReadOnlySpan<byte> valueSpan = json.ValueSpan;
-				switch (tokenType) {
-				case JsonTokenType.StartObject:
-				case JsonTokenType.EndObject:
-				case JsonTokenType.Null:
-				case JsonTokenType.StartArray:
-				case JsonTokenType.EndArray:
-					break;
-				case JsonTokenType.PropertyName:
-					if (valueSpan.SequenceEqual(StreamsKeyword)) {
-						currentObject = JsonObjects.Streams;
-						break;
-					}
-					if (valueSpan.SequenceEqual(FormatKeyword)) {
-						currentObject = JsonObjects.Format;
-						break;
+					if (stream.r_frame_rate != null) {
+						if (stream.r_frame_rate.Contains('/')) {
+							var split = stream.r_frame_rate.Split('/');
+							if (split.Length == 2 && int.TryParse(split[0], out var firstRate) && int.TryParse(split[1], out var secondRate))
```

Comment:
Use explicit types. Why do you check for split length to be 2 when you are using only the first one?

Reply:
*smirk*
This is your code, see your version, line 177.
>  check for split length to be 2 
Because in the last part of the line,` int.TryParse(split[1], out var secondRate)` is used, right?

Reply:
Sorry I missed this one, you're right.